test/integration/parachain/staging/sequential/asset-registry.test.ts

```diff
@@ -0,0 +1,126 @@
+import { assert } from "../../../../chai";
+import { ApiPromise, Keyring } from "@polkadot/api";
+import { KeyringPair } from "@polkadot/keyring/types";
+import { createSubstrateAPI } from "../../../../../src/factory";
+import { ESPLORA_BASE_PATH, PARACHAIN_ENDPOINT, SUDO_URI } from "../../../../config";
+import { DefaultAssetRegistryAPI, DefaultInterBtcApi, getStorageKey, stripHexPrefix } from "../../../../../src";
+
+import { StorageKey } from "@polkadot/types";
+import { AnyTuple } from "@polkadot/types/types";
+
+import { OrmlAssetRegistryAssetMetadata } from "@polkadot/types/lookup";
+import { waitForFinalizedEvent } from "../../../../utils/helpers";
+
+describe("AssetRegistry", () => {
+    let api: ApiPromise;
+    let interBtcAPI: DefaultInterBtcApi;
+
+    let sudoAccount: KeyringPair;
+
+    let assetRegistryMetadataHash: string;
+    let registeredKeysBefore: StorageKey<AnyTuple>[] = [];
+
+    before(async () => {
+        const keyring = new Keyring({ type: "sr25519" });
+        api = await createSubstrateAPI(PARACHAIN_ENDPOINT);
+
+        sudoAccount = keyring.addFromUri(SUDO_URI);
+        interBtcAPI = new DefaultInterBtcApi(api, "regtest", sudoAccount, ESPLORA_BASE_PATH);
+
+        assetRegistryMetadataHash = getStorageKey("AssetRegistry", "Metadata");
+        // check which keys exist before the tests
+        registeredKeysBefore = (await interBtcAPI.api.rpc.state.getKeys(assetRegistryMetadataHash)).toArray();
+    });
+    
+    after(async () => {
+        // clean up keys created in tests if necessary
+        const registeredKeysAfter = (await interBtcAPI.api.rpc.state.getKeys(assetRegistryMetadataHash)).toArray();
+
+        const previousKeyHashes = registeredKeysBefore.map(key => stripHexPrefix(key.toHex()));
+        // need to use string comparison since the raw StorageKeys don't play nicely with .filter()
+        const newKeys = registeredKeysAfter.filter(key => {
+            const newKeyHash = stripHexPrefix(key.toHex());
+            return !previousKeyHashes.includes(newKeyHash);
+        });
+
+        if (newKeys.length > 0) {
+            // clean up assets registered in test(s)
+            const deleteKeysInstruction = interBtcAPI.api.tx.system.killStorage(newKeys);
+            await interBtcAPI.api.tx.sudo.sudo(deleteKeysInstruction).signAndSend(sudoAccount);
+
+            // wait for finalized event
+            await waitForFinalizedEvent(interBtcAPI, api.events.sudo.Sudid);
```

Comment:
I was about to suggest using `DefaultTransactionAPI.sendLogged(api, sudoAccount, interBtcAPI.api.tx.sudo.sudo(deleteKeysInstruction), api.events.sudo.Sudid)`, but then noticed that for some reason the `successEventType` parameter is [not used ](https://github.com/interlay/interbtc-api/blob/d8446ca0eb272af29212ec646df4531d1e7800da/src/parachain/transaction.ts#L51). Specifying that parameter used to check its existence in the `result.events` [array](https://github.com/interlay/interbtc-api/blob/d8446ca0eb272af29212ec646df4531d1e7800da/src/parachain/transaction.ts#L84). For another time, but we should add that check back.

Reply:
Yeah, this was a bit of an awkward one. Would be good to at least check if the sudo was successful, but in this case, I hope we end up with dead code as soon as our genesis state contains a few foreign assets - which means I do not need to create them, and therefore, not delete them with those awkward sudos.

Reply:
I noticed you send the tx and then `waitForFinalizedEvent` in some other places too, so it would still be good to replace that by updating the `sendLogged(...)` method